src/hotspot/cpu/ppc/gc/shenandoah/shenandoahBarrierSetAssembler_ppc.cpp

```diff
@@ -0,0 +1,1017 @@
+    // |
+    // |------------------------------------------------------  <--- cs base address + (heap_base >> region size shift)
+    //                                                                               + number of regions
+    __ load_const_optimized(tmp2, ShenandoahHeap::in_cset_fast_test_addr(), tmp1);
+    __ srdi(tmp1, dst, ShenandoahHeapRegion::region_size_bytes_shift_jint());
+    __ lbzx(tmp2, tmp1, tmp2);
+    __ andi_(tmp2, tmp2, 1);
+    __ beq(CCR0, skip_barrier);
+  }
+
+  /* ==== Invoke runtime ==== */
+  // Save to-be-preserved registers.
+  int nbytes_save = 0;
+
+  const bool needs_frame           = preservation_level >= MacroAssembler::PRESERVATION_FRAME_LR;
+  const bool preserve_gp_registers = preservation_level >= MacroAssembler::PRESERVATION_FRAME_LR_GP_REGS;
+  const bool preserve_fp_registers = preservation_level >= MacroAssembler::PRESERVATION_FRAME_LR_GP_FP_REGS;
+
+  if (needs_frame) {
+    if (preserve_gp_registers) {
+      nbytes_save = (preserve_fp_registers
+                     ? MacroAssembler::num_volatile_gp_regs + MacroAssembler::num_volatile_fp_regs
+                     : MacroAssembler::num_volatile_gp_regs) * BytesPerWord;
+      __ save_volatile_gprs(R1_SP, -nbytes_save, preserve_fp_registers);
+    }
+
+    __ save_LR_CR(tmp1);
+    __ push_frame_reg_args(nbytes_save, tmp1);
+  }
+
+  // Calculate the reference's absolute address.
+  __ add(R4_ARG2, ind_or_offs, base);
+
+  // Invoke runtime.
+  address jrt_address = nullptr;
+
+  if (is_strong) {
+    if (is_narrow) {
+      jrt_address = CAST_FROM_FN_PTR(address, ShenandoahRuntime::load_reference_barrier_strong_narrow);
+    } else {
+      jrt_address = CAST_FROM_FN_PTR(address, ShenandoahRuntime::load_reference_barrier_strong);
+    }
+  } else if (is_weak) {
+    if (is_narrow) {
+      jrt_address = CAST_FROM_FN_PTR(address, ShenandoahRuntime::load_reference_barrier_weak_narrow);
+    } else {
+      jrt_address = CAST_FROM_FN_PTR(address, ShenandoahRuntime::load_reference_barrier_weak);
+    }
+  } else {
+    assert(is_phantom, "only remaining strength");
+    assert(!is_narrow, "phantom access cannot be narrow");
+    jrt_address = CAST_FROM_FN_PTR(address, ShenandoahRuntime::load_reference_barrier_phantom);
+  }
+  assert(jrt_address != nullptr, "jrt routine cannot be found");
+
+  __ call_VM_leaf(jrt_address, dst /* reference */, R4_ARG2 /* reference address */);
+
+  // Restore to-be-preserved registers.
+  if (preserve_gp_registers) {
+    __ mr(R0, R3_RET);
+  } else {
+    __ mr_if_needed(dst, R3_RET);
+  }
+
+  if (needs_frame) {
+    __ pop_frame();
+    __ restore_LR_CR(tmp1);
+
+    if (preserve_gp_registers) {
+      __ restore_volatile_gprs(R1_SP, -nbytes_save, preserve_fp_registers);
+      __ mr(dst, R0);
+    }
+  }
+
+  __ bind(skip_barrier);
+
+#ifdef ASSERT
+  if (!preserve_gp_registers) { __ clobber_volatile_gprs(dst); }
+  if (!needs_frame) { __ clobber_carg_stack_slots(tmp1); }
+#endif
```

Comment:
This clobber code was certainly good during development and early testing. But is it worth keeping it? Other GCs and other places don't have it any more. So, I'd slightly prefer removal. Feel free to do so if you agree.